### PR TITLE
Removed the need of a global config at plugin loading time

### LINKED ISSUE #16: snap-plugin-collector-elasticsearch will not load with snapctl plugin load
Problem: Built snap-plugin-collector-elasticsearch from latest [commit](https://github.com/intelsdi-x/snap-plugin-collector-elasticsearch/commit/821fba91323f8766c1d48274208891cb0f89f229
)
When attempting to load snap-plugin-collector-elasticsearch into snap with snapctl plugin load I receive the following error:
```
<029-zwzca:/go/bin# snapctl plugin load snap-plugin-collector-elasticsearch
Error loading plugin:
unexpected EOF
```

Environment:
- snap in tribe mode in kubernetes on gce
- snap plugins loaded:
```
NAME             VERSION         TYPE            SIGNED          STATUS          LOADED TIME
heapster         1               publisher       false           loaded          Fri, 09 Sep 2016 17:09:50 UTC
docker           8               collector       false           loaded          Fri, 09 Sep 2016 17:09:56 UTC
cassandra        3               collector       false           loaded          Fri, 09 Sep 2016 17:10:01 UTC
passthru         1               processor       false           loaded          Fri, 09 Sep 2016 17:09:41 UTC
influx           15              publisher       false           loaded          Fri, 09 Sep 2016 17:09:46 UTC
```

### PR DESCRIPTION
**Pull request by candysmurf**
#### Summary of changes  fix #16 
* no global config is needed at the plugin loading time
* change the steps on README

#### Tests done
*  unit tests
* integration tests
* loading the plugins using task manifest

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -221,25 +221,26 @@ Create a task manifest file (e.g. `elasticsearch-task.json`).
...
             },
-            "config": {},
             "process": [
                 {
                     "plugin_name": "passthru",
                     "process": null,
                     "publish": [
                         {                         
-                            "plugin_name": "file",
+                            "plugin_name": "mock-file",
```
Comment (jcooklin) [inline README.md]: In general we shouldn't point users to mock plugins.  Mocks are used to test the framework.

File: elasticsearch/elasticsearch.go [hunk 2]
```diff
@@ -45,23 +48,26 @@ const (
 	esPort = "port"
 
 	invalidMetricType = "Invalid metric type found"
+	readDocErr        = "Cannot read the file"
```
Comment (jcooklin) [inline elasticsearch/elasticsearch.go]: This is an unexported err that is referenced only once.  For readability I would move this to where the error is thrown and maybe include more than "cannot read the file".

**Comment by candysmurf** [comment]
@jcooklin, incorporated feedback. Please take another look. thanks.

**Comment by jcooklin** [comment]
LGTM

**Pull request merged**